pkg/skaffold/runner/new.go

```diff
@@ -188,10 +188,15 @@ func isImageLocal(runCtx *runcontext.RunContext, imageName string) (bool, error)
 
 	cl := runCtx.GetCluster()
 	var pushImages bool
-	if pipeline.Build.LocalBuild.Push == nil {
+
+	switch {
+	case runCtx.Opts.PushImages.Value() != nil:
+		logrus.Debugf("push value set via skaffold build --push flag, --push=%t", *runCtx.Opts.PushImages.Value())
+		pushImages = *runCtx.Opts.PushImages.Value()
+	case pipeline.Build.LocalBuild.Push == nil:
 		pushImages = cl.PushImages
 		logrus.Debugf("push value not present, defaulting to %t because cluster.PushImages is %t", pushImages, cl.PushImages)
-	} else {
+	default:
 		pushImages = *pipeline.Build.LocalBuild.Push
```

Comment:
Can we add tests here so we never unintentionally break the precedence?
Test to make sure if `runCtx.Opts.PushImages` is not set a and pipeline.Build.LocalBuild.Push is set, then `pushImages` is true. 

If ``runCtx.Opts.PushImages` is set to false and `pipeline.Build.LocalBuild.Push` is true, pushImages is set to false?

Reply:
I'll add this, thanks!  Do you have any code examples of how to mock/set skaffold.yaml values or directly set the pipelines values?  Still trying to understand how to do this from poking around.

Reply:
I think you can get away with just testing `isImageLocal` function with manually set values for `runcontext.RunContext`?

Reply:
Ah makes sense, added `TestIsImageLocal` to new_test.go testing all combinations of --push=[nil|false|true] and pipeline.Build.LocalBuild.Push=[nil|false|true] w/ desired functionality